src/model/FaceliftModel.h

```diff
@@ -1000,9 +1001,133 @@ class Model : public ModelBase
 
 };
 
+
+template<typename Type, typename Sfinae = void>
+struct QMLModelTypeHandler
+{
+    static QJSValue toJSValue(const Type &v, QQmlEngine *engine)
+    {
+        return engine->toScriptValue(facelift::toQMLCompatibleType(v));
+    }
+
+    static void fromJSValue(Type &v, const QJSValue &value, QQmlEngine *engine)
+    {
+        v = engine->fromScriptValue<Type>(value);
+    }
+
+};
+
+
+template<typename Type>
+QJSValue toJSValue(const Type &v, QQmlEngine *engine)
+{
+    return QMLModelTypeHandler<Type>::toJSValue(v, engine);
 }
 
 
+template<typename Type>
+void fromJSValue(Type &v, const QJSValue &jsValue, QQmlEngine *engine)
+{
+    QMLModelTypeHandler<Type>::fromJSValue(v, jsValue, engine);
+}
+
+
+class AsyncAnswerBase
+{
+
+public:
+    AsyncAnswerBase()
+    {
+    }
+
+};
+
+template<typename ReturnType>
+class AsyncAnswer : public AsyncAnswerBase
+{
+    typedef std::function<void (const ReturnType &)> CallBack;
+
+public:
+    struct Master
+    {
+
+        Master(CallBack callback) : m_callback(callback)
+        {
+        }
+
+        ~Master()
+        {
+            if (!m_isAlreadyAnswered) {
+                qWarning() << "No answer provided to asynchronous call";
+            }
+        }
+
+        bool m_isAlreadyAnswered = false;
+        CallBack m_callback;
+    };
+
+    AsyncAnswer()
```

Comment:
can be removed?

Reply:
there must be a constructor with no parameter to enable usage in QMap